src/addons/plugin/device-proxy.js

```diff
@@ -27,8 +27,21 @@ class DeviceProxy extends Device {
       this.properties.set(propertyName, propertyProxy);
     }
 
+    // Copy over any extra device fields which might be useful for debugging.
+    this.deviceDict = {};
+    for (let field in deviceDict) {
+      if (field in ['id', 'name', 'type', 'properties', 'actions', 'events']) {
+        continue;
+      }
+      this.deviceDict[field] = deviceDict[field];
+    }
+
     //TODO: Add support for actions once we know what they look like.
   }
+
+  asDict() {
+    return Object.assign({}, this.deviceDict, super.asDict());
```

Comment:
This is going to overlay the base class properties on top of this object's properties. Was that intended?

Reply:
I went back and forth on this. I think that having the properties as a separate dict is a better solution, but it requires a bunch more changes. Let's defer this to a separate issue.

Reply:
Although, the extra stuff that's being merged is the same fields that the derived class has in its own definition. So there should never be a collision (unless there is already a collision in the derived class). The Proxy classes just flatten it out into a single object instead of being spread out via inheritance.